nixos/lib/test-driver/test-driver.py

```diff
@@ -655,6 +655,27 @@ def wait_for_x(self):
                 if status == 0:
                     return
 
+    def get_window_names(self):
+        return self.succeed(
+            r"xwininfo -root -tree | sed 's/.*0x[0-9a-f]* \"\([^\"]*\)\".*/\1/; t; d'"
```

Comment:
can we not use `xwininfo`? I'm sure there's a python library just for this.

Reply:
hey @worldofpeace, thank you for the suggestion. It surely makes sense to use better alternatives if there are any, but questioning the methods of the perl scripts, researching what other libraries are there and doing things differently for all the 300 tests that have to be ported significantly blows up the amount of work we have to do here.

Can we please discuss such things offline and find consensus on describing in follow-up tickets on how to improve such things once the port is complete - or find people who like to help out on such things?